### PR TITLE
feat: add an option to provide nameselector and scope in operator webhook

### PR DESCRIPTION
**Pull request by ypoplavs**
## Pull request description 
As per user's [request](https://github.com/kubeshop/helm-charts/issues/760) this PR enabled adding `namespaceSelector` to limit which requests are intercepted.

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by vsukhin — APPROVED** [review]

File: charts/testkube-operator/templates/webhook.yaml [hunk 1]
```diff
@@ -3,6 +3,10 @@ apiVersion: admissionregistration.k8s.io/v1
 kind: ValidatingWebhookConfiguration
 metadata:
   name: {{ .Release.Name }}-{{ .Values.webhook.name }}
+  {{- if .Values.webhook.namespaceSelector }}
```
Comment (dejanzele) [inline charts/testkube-operator/templates/webhook.yaml:6]: this should go in the `webhooks` section, not in `metadata`

**Review by dejanzele — CHANGES_REQUESTED** [review]

File: charts/testkube-operator/values.yaml [hunk 2]
```diff
@@ -146,6 +146,10 @@ webhook:
   enabled: true
   ## Name of the webhook
   name: webhook-admission
+  ## Limit which requests for namespaced resources are intercepted, by specifying a namespaceSelector.
+  namespaceSelector: []
```
Comment (dejanzele) [inline charts/testkube-operator/values.yaml:150]: default value is `{}` not an array `[]`

File: charts/testkube-operator/templates/webhook.yaml [hunk 3]
```diff
@@ -40,5 +44,8 @@ webhooks:
     - UPDATE
     resources:
     - testtriggers
+    {{- if .Values.webhook.scope }}
+    scope: {{ .Values.webhook.scope }}
```
Comment (dejanzele) [inline charts/testkube-operator/templates/webhook.yaml:48]: `scope` can be hardcoded to `Namespaced` as all our resources are namespace-scoped

File: charts/testkube-operator/values.yaml [hunk 4]
```diff
@@ -146,6 +146,10 @@ webhook:
...
+  ## Scope specifies a scope to match. Valid values are "Cluster", "Namespaced", and "*". Default is "*".
+  scope: ""
```
Comment (dejanzele) [inline charts/testkube-operator/values.yaml:152]: check my comment for hardcoding, this won't be needed if it is hardcoded to Namespaced

File: charts/testkube/values.yaml [hunk 5]
```diff
@@ -1157,6 +1157,10 @@ testkube-operator:
     enabled: true
     # -- Name of the webhook
     name: testkube-operator-webhook-admission
+    ## Limit which requests for namespaced resources are intercepted, by specifying a namespaceSelector.
+    namespaceSelector: []
+    ## Scope specifies a scope to match. Valid values are "Cluster", "Namespaced", and "*". Default is "*".
+    scope: ""
```
Comment (dejanzele) [inline charts/testkube/values.yaml:1163]: same as above

File: charts/testkube/values.yaml [hunk 6]
```diff
@@ -1157,6 +1157,10 @@ testkube-operator:
     enabled: true
     # -- Name of the webhook
     name: testkube-operator-webhook-admission
+    ## Limit which requests for namespaced resources are intercepted, by specifying a namespaceSelector.
+    namespaceSelector: []
```
Comment (dejanzele) [inline charts/testkube/values.yaml:1161]: same as above

File: charts/testkube-operator/templates/webhook.yaml [hunk 7]
```diff
@@ -3,6 +3,10 @@ apiVersion: admissionregistration.k8s.io/v1
 kind: ValidatingWebhookConfiguration
 metadata:
   name: {{ .Release.Name }}-{{ .Values.webhook.name }}
+  {{- if .Values.webhook.namespaceSelector }}
+  namespaceSelector:
+  {{ toYaml .Values.webhook.namespaceSelector | nindent 4 | trim | indent 2 }}
```
Comment (dejanzele) [inline charts/testkube-operator/templates/webhook.yaml:8]:
```suggestion
  {{- toYaml .Values.webhook.namespaceSelector | nindent 4 | trim | indent 2 }}
```

**Pull request merged**